src/client/providers/queryTransformer.ts

```diff
@@ -0,0 +1,39 @@
+import { Observable, of } from 'rxjs'
+import { flatMap, map, switchMap } from 'rxjs/operators'
+import { FeatureProviderRegistry } from './registry'
+
+export type TransformQuerySignature = (query: string) => Observable<string | null | undefined>
+
+/** Provides transformed queries from the first search extension. */
+export class QueryTransformerRegistry extends FeatureProviderRegistry<{}, TransformQuerySignature> {
+    public transformQuery(query: string): Observable<string | null | undefined> {
+        return transformQuery(this.providers, query)
+    }
+}
+
+/**
+ * Returns an observable that emits a query transformed by all providers whenever any of the last-emitted set of providers emits
+ * a query.
+ *
+ * Most callers should use QueryTransformerRegistry's transformQuery method, which uses the registered search
```

Comment:
The term "search provider" is inaccurate; there is no such concept (or it is an imprecise way to refer to query transformers). It should be "query transformers".